### PR TITLE
Multi serial tilelink

### PR DESCRIPTION
**Pull request by jerryz123**
Support multiple serial tilelink links

### CODE REVIEW

**Comment by kevindna** [comment]
General comment is that you added a  couple new configs, but no comments about why? Like I _could_ search through the repo, OR I could read a one liner. Rejected

**Comment by jerryz123** [comment]
Could you point to the configs for which you'd like a in-line comment explaining the use case?

**Comment by kevindna** [comment]
I will list line numbers since some are not full configs: 
- _src/main/scala/PeripheryTLSerial.scala_: Line 19
- _src/main/scala/PeripheryTLSerial.scala_: Line 23
- _src/main/scala/Configs.scala_: Line 141 (How should I use this?)
- _src/main/scala/Configs.scala_: Line 199 (coherent TLB??)
- _src/main/scala/Configs.scala_: Liine 106 (this does __)
- _src/main/scala/TSIHarness.scala_: Line 199 ("Iterate over TL connections")

File: src/main/scala/Configs.scala [hunk 1]
```diff
@@ -61,87 +61,103 @@ class WithNBlockDeviceTrackers(n: Int) extends Config((site, here, up) => {
...
+  case SerialTLKey => up(SerialTLKey).map(_.copy(bundleParams=params))
+})
 
-class WithSerialTLBackingMemory extends Config((site, here, up) => {
-  case ExtMem => None
-  case SerialTLKey => up(SerialTLKey, site).map { k => k.copy(
-    serialTLManagerParams = Some(k.serialTLManagerParams.getOrElse(SerialTLManagerParams(memParams = up(ExtMem).get.master))
-      .copy(memParams = up(ExtMem).get.master, isMemoryDevice = true))
-  )}
+class WithSerialTLBCE extends Config((site, here, up) => {
```
Comment (kevindna) [inline src/main/scala/Configs.scala:119]: One linear please. Better not leave it ambiguous

**Comment by jerryz123** [comment]
> I will list line numbers since some are not full configs:
> 
> * _src/main/scala/PeripheryTLSerial.scala_: Line 19
> * _src/main/scala/PeripheryTLSerial.scala_: Line 23
> * _src/main/scala/Configs.scala_: Line 141 (How should I use this?)
> * _src/main/scala/Configs.scala_: Line 199 (coherent TLB??)
> * _src/main/scala/Configs.scala_: Liine 106 (this does __)
> * _src/main/scala/TSIHarness.scala_: Line 199 ("Iterate over TL connections")

I've addressed most of these, but I couldn't find what you are referring to in TSIHarness.scala (where is line 199?). Use inline comments next time please.

File: src/main/scala/Configs.scala (hunk 1 above)
Comment (jerryz123) [reply src/main/scala/Configs.scala:140]: You mean you prefer this as a one-liner?

Comment (kevindna) [reply src/main/scala/Configs.scala:140]: I meant to add a one line comment

Comment (jerryz123) [reply src/main/scala/Configs.scala:140]: Ok, I did that in the most recent commits.

**Review by kevindna — COMMENTED** [review]
Great example of how one line comment bring quick clarity and inform use. Approved!

**Review by kevindna — APPROVED** [review]

**Comment by jerryz123** [comment]
Thanks for the review! I'll merge this together with the Chipyard one once that passes tests

**Pull request merged**